src/Microsoft.Azure.WebJobs.Host/Tables/CollectorArgumentBindingProvider.cs

```diff
@@ -22,8 +23,8 @@ public IStorageTableArgumentBinding TryCreate(ParameterInfo parameter)
             }
 
             Type entityType = GetCollectorItemType(parameter.ParameterType);
-
-            if (!TableClient.ImplementsOrEqualsITableEntity(entityType))
+                        
+            if (!TableClient.ImplementsOrEqualsITableEntity(entityType) && !TypeUtility.IsJObject(entityType))
             {
```

Comment:
Not necessarily for this PR, but could this check be moved to invocation-time? The other day I was trying to use an anonymous type that did have PK and RK properties and it was rejected. It'd be cool if that was enabled.